File: PA6Java/src/pa6/ExceptionHandler.java
package pa6;

public class ExceptionHandler {
	
	// Method for handling exceptions
	public String HandleException(Exception ex) {
		
		// Error message for connection
		if (ex.getMessage().contains("error: 40 "))
		{
			return "Check your internet connection" +  "\n or contact our IT support at: 0730880819";
		}
		// Error message for failing to access database
		else if (ex.getMessage().contains("Cannot open database"))
		{
			return "Connection failed to the database," + "\n please check if the name is spelled correctly" +"\n for the database or if the username" +"\n and/or password is correct," + "\n or if the user has access to the database";
		}
		// Error message for server connection
		else if (ex.getMessage().contains("error: 26"))
		{
			return "Connection failed to the server please" + "\n check if you are connected to the server.";
		}
		// Error message for unexpected error
		else
        {
            return "Unknown error contact our IT support at: 0730880819";
        }
	}
}

File: PA2/ExceptionHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PA2
{
	class ExceptionHandler
	{

		// Method for handling our exceptions and returns a string dependent on what the errormessage contains

		public string HandleException(Exception ex)
		{
            // Error message if there is no connection

            if (ex.Message.Contains("error: 40 "))
			{
				return "Check your internet connection or contact our IT support at: 0730880819";
			}

            // Error message for failing to access database

            else if (ex.Message.Contains("Cannot open database"))
			{
				return "Connection failed to the database, please check if the name is spelled correctly for the database or if the username and/or password is correct, or if the user has access to the database";
			}

            // Error message for server connection

            else if (ex.Message.Contains("error: 26"))
			{
				return "Connection failed to the server please check if you are connected to the server.";
			}

            //Error messege when unknown error

			else
			{
				return "Try to restart the program if it is still not working please call our customer service at: 0730880819";
			}
		}
	}
}
